src/arrayMethodJoin.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let final = '';
+    let joinSeparator = separator;
+
+    if (joinSeparator === undefined) {
+      joinSeparator = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
```

Comment:
just skip redundant iteration
```suggestion
      if (element === null || element === undefined) {
        break;
      }
```

Reply:
I don't pass the tests if I skip this iteration